File: modules/wem-api/src/main/java/com/enonic/wem/query/expr/DynamicOrderExpr.java
package com.enonic.wem.query.expr;

public final class DynamicOrderExpr
    extends OrderExpr
{
    private final FunctionExpr function;

    public DynamicOrderExpr( final FunctionExpr function, final Direction direction )
    {
        super( direction );
        this.function = function;
    }

    public FunctionExpr getFunction()
    {
        return this.function;
    }

    @Override
    public String toString()
    {
        return this.function + " " + getDirection();
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/query/expr/DynamicOrderExpr.ts
module api.query.expr {

    export class DynamicOrderExpr extends OrderExpr {

        private func: FunctionExpr;

        constructor(func: FunctionExpr, direction: OrderDirection) {
            super(direction);
            this.func = func;
        }

        getFunction(): FunctionExpr {
            return this.func;
        }

        toString() {
            return this.func.toString() + " " + super.directionAsString();
        }
    }
}
